Change more state references

arscan opened the pull request:
More state reference changes.

kjmahalingam commented:
The `PriorState` conditional also needs this functionality, as it references states as well.

dehall commented on src/reducers/modules.js:
```diff
@@ -55,7 +65,45 @@ const renameModuleState = (module, stateName, newName) => {
+        break;
+      case 'Death':
+      case 'ConditionEnd':
+        if(state.condition_onset === stateName){
```
There should also be an 'AllergyEnd' with an `allergy_onset` property to check here. I think this covers all the rest of them though.

I also think you may not need the switch at all, assuming checking `state.reason === stateName` won't crash if the state doesn't have a `reason` field. That way if we add new states with a `reason` field, or add a `reason` field to another existing state, we won't have to change this. We've been trying to be consistent such that the same field name means the same thing on all states, so a `reason` , `condition_onset`, etc, will always be a reference to another state.

arscan commented:
Ok, I added allergy_onset, and removed the switch statement, assuming that you do not reuse attribute names to mean different things.  Also found that target_encounter is required, so instead of deleting that field on deletion, I set it to "".  Should be good to test and merge.  I think this should be tested fairly thoroughly because it touches so many states and can result in a hard crash if I misunderstood what is required and what isn't.
